grr/server/grr_response_server/gui/ui/lib/api_translation/client.ts

```diff
@@ -122,26 +126,32 @@ function createOptionalUnixVolume(volume?: ApiUnixVolume): UnixVolume | undefine
 }
 
 function createStorageVolume(apiVolume: ApiVolume): StorageVolume {
-  let totalSize = 0;
-  let freeSpace = 0;
+  let totalSize = undefined;
+  let freeSpace = undefined;
+  let bytesPerSector = undefined;
+
   if (apiVolume.bytesPerSector && apiVolume.sectorsPerAllocationUnit) {
     if (apiVolume.totalAllocationUnits) {
-      totalSize = apiVolume.bytesPerSector *
-          apiVolume.sectorsPerAllocationUnit * apiVolume.totalAllocationUnits;
+      totalSize = BigInt(apiVolume.bytesPerSector) *
+          BigInt(apiVolume.sectorsPerAllocationUnit) * BigInt(apiVolume.totalAllocationUnits);
     }
 
     if (apiVolume.actualAvailableAllocationUnits) {
-      freeSpace = apiVolume.bytesPerSector *
-          apiVolume.sectorsPerAllocationUnit *
-          apiVolume.actualAvailableAllocationUnits;
+      freeSpace = BigInt(apiVolume.bytesPerSector) *
+          BigInt(apiVolume.sectorsPerAllocationUnit) *
+          BigInt(apiVolume.actualAvailableAllocationUnits);
     }
   }
 
+  if (apiVolume.bytesPerSector) {
```

Comment:
Implicit testing for truthyness with `if (variable)` can lead to unexpected results. As an example, the `if` branch is executed for `"0"`, but not for `""` or `0`. Please use `===` instead.

See: https://dorey.github.io/JavaScript-Equality-Table/

Reply:
I'll check explicitly for undefined values